src/wifi.cpp

```diff
@@ -0,0 +1,694 @@
+int ip_socket_file_descriptor = 0;
+// Keep track of the handle of the task that reads IP packets (task_read_ip_packets(...))
+TaskHandle_t read_ip_packet_task_handle = nullptr;
+// Keep track of the esp_netif object attaching netif to wifi and registering wifi handlers to the default event loop
+void *esp_netif = nullptr;
+
+// Keep track of the special strings that if a TCP packet matches,
+// should excute an action, and what action they should ewxecute
+tcp_command_t tcp_commands[NUM_TCP_COMMANDS] = {
+    {
+        .command = "up",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_UP,
+            /* bool from_isr = */ false); },
+    },
+    {
+        .command = "down",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_DOWN,
+            /* bool from_isr = */ false); },
+    },
+    {
+        .command = "confirm",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_CONFIRM,
+            /* bool from_isr = */ false); },
+    },
+#if 0
+    {
+        .command = "sleep",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_SLEEP,
+            /* bool from_isr = */ false); },
+    },
+#endif
+};
+
+// ============ //
+// Define tasks //
+// ============ //
+
+// A task whose job it is to read all incoming TCP packets over the connected IP socket.
+// Based on the contents of the TCP packet, it may execute certain actions.
+// Once the socket the TCP connection is set up on, ip_socket_file_descriptor,
+// is no longer readable, it automatically kills itself and resets the handle.
+void task_read_ip_packets()
+{
+    // Create a 0-initialized buffer IP packets will be read into
+    char read_buffer[16] = { 0 };
+    int num_read_bytes = 0;
+
+    while(1)
+    {
+        // Get the number of bytes in the file descriptor,
+        // copy up to count bytes from the file descriptor to our read buffer
+        // https://man7.org/linux/man-pages/man2/read.2.html
+        num_read_bytes = read(
+            /* int fd = */ ip_socket_file_descriptor,
+            /* void buf[.count] = */ read_buffer,
+            /* size_t count = */ sizeof(read_buffer) - sizeof('\n'));
+        if(0 > num_read_bytes)
+        {
+            // Failed to read the file descriptor, stop this task and free its resources
+            s_println("Failed to read IP packet file descriptor, stopping task to read IP packets");
+            (void) close(/* int fd = */ ip_socket_file_descriptor);
+            ip_socket_file_descriptor = 0;
+            read_ip_packet_task_handle = nullptr;
+            vTaskDelete(/* TaskHandle_t xTaskToDelete = */ NULL);
+        }
+
+        // Print the bytes from our user input to stdout (replace ending \n with \0)
+        read_buffer[num_read_bytes - sizeof('\n')] = '\0';
+        s_println(read_buffer);
```

Comment:
Improve this print statement to says what it's printing or remove it.

Reply:
addressed in: https://github.com/jCallon/squirt/pull/1/commits/edb88a636d2cf54a42f970d190cd054c04993412